RecoTauTag/HLTProducers/interface/PFDiJetCorrCheckerWithDiTau.h

```diff
@@ -0,0 +1,37 @@
+/** Description: Check correlation between PFJet pairs and filtered PFTau pairs and store the PFJet pairs.
+For (j1, j2, t1, t2) where j1, j2 from the PFJet collection and t1, t2 from the filtered PFTau collection,
+the module checks if there is no overlap (within dRmin) between j1, j2, t1, t2, i.e. they are 4 different objects.
+In addition, the module imposes the following cuts:
+* mjjMin: the min invariant mass cut on (j1, j2)
+* extraTauPtCut: the leading tau pt cut on (t1, t2) (under the assumption t1, t2 are products of a subleading pt filter with minN = 2)
+The module stores j1, j2 of any (j1, j2, t1, t2) that satisfies the conditions above. */
+
+#ifndef RecoTauTag_HLTProducers_PFDiJetCorrCheckerWithDiTau_H
+#define RecoTauTag_HLTProducers_PFDiJetCorrCheckerWithDiTau_H
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+
+class PFDiJetCorrCheckerWithDiTau : public edm::stream::EDProducer<> {
+private:
+  const edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs> tauSrc_;
+  const edm::EDGetTokenT<reco::PFJetCollection> pfJetSrc_;
+  const double extraTauPtCut_;
+  const double mjjMin_;
+  const double matchingR2_;
+
+public:
+  explicit PFDiJetCorrCheckerWithDiTau(const edm::ParameterSet&);
+  ~PFDiJetCorrCheckerWithDiTau() override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+};
+#endif
```

Comment:
Please combine .h and .cc file into one file, and add there the FWK macro call.

Reply:
OK I will do. Where such merged file should be placed, into src directory? This package has not plugins directory, however it can be created if needed.

Reply:
So far for this pkg, `src/` was used (`HLTPFTauIPFilter` is an example).

Unrelated: I was going to mention that HLT plugins usually start with "HLT" in their name, but I see that other plugins in this pkg are not following that convention. Is there a choice or need behind this? I'm just curious; I don't have a strong opinion on the name of the plugin in this PR.

Reply:
If it is used only at HLT, then starting with "HLT" would be good. If it is of more general use, also in RECO, then either way is fine.

Reply:
> So far for this pkg, `src/` was used (`HLTPFTauIPFilter` is an example).

Thanks

> Unrelated: I was going to mention that HLT plugins usually start with "HLT" in their name, but I see that other plugins in this pkg are not following that convention. Is there a choice or need behind this? I'm just curious; I don't have a strong opinion on the name of the plugin in this PR.

Filter names in this package start from "HLT" while names of producers not if I am not mistaken. But I do not think there is any strict naming convention behind this.

Reply:
> If it is used only at HLT, then starting with "HLT" would be good. If it is of more general use, also in RECO, then either way is fine.

The module under discussion uses refs to HLT-filter objects, so it cannot be easily used in RECO. Therefore, I will add HLT at beginning of its name.